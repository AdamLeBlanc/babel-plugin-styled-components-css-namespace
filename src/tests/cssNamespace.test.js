import pluginTester from 'babel-plugin-tester';
import plugin from '../index';
import path from 'path';

pluginTester({
  plugin,
  pluginName: 'styled-components-css-namespace',
  filename: __filename,
  // fixtures: path.join(__dirname, './fixtures'),
  snapshot: true,
  tests: [
    {
      title: 'Adds namespace to simple styled-component',
      fixture: path.join(__dirname, './fixtures/simple_styled_component.js')
    },
    {
      title: 'Adds namespace to complex styled-component',
      fixture: path.join(__dirname, './fixtures/complex_styled_component.js')
    },
    {
      title: 'Uses a namespace specified in the options',
      pluginOptions: {
        cssNamespace: 'moreSpecific'
      },
      fixture: path.join(__dirname, './fixtures/complex_styled_component.js')
    },
    {
      title:
        'Does not transform a template string that is not a styled component',
      snapshot: false, // snapshot should be false because code should be unmodified
      fixture: path.join(__dirname, './fixtures/not_styled_component.js')
    }
  ]
});
