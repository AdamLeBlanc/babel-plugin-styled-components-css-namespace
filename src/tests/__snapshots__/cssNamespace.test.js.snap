// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds namespace to complex styled-component 1`] = `
"
const styled = { div() {} };

const MyStyledComponent = styled.div\`
  background-color: yellow;
  color: {props => props.theme.color};
  
  .foo {
    border: 1px solid black;

    & .bar {
      background-color: blue;
    }
  }
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

const styled = { div() {} };

const MyStyledComponent = styled.div\`
&& {
  background-color: yellow;
  color: {props => props.theme.color};
  
  .foo {
    border: 1px solid black;

    & .bar {
      background-color: blue;
    }
  }
}
\`;
"
`;

exports[`Adds namespace to simple styled-component 1`] = `
"
const styled = { div() {} };

const MyStyledComponent = styled.div\`
  background-color: 'yellow';
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

const styled = { div() {} };

const MyStyledComponent = styled.div\`
&& {
  background-color: 'yellow';
}
\`;
"
`;

exports[`Uses a namespace specified in the options 1`] = `
"
const styled = { div() {} };

const MyStyledComponent = styled.div\`
  background-color: yellow;
  color: {props => props.theme.color};
  
  .foo {
    border: 1px solid black;

    & .bar {
      background-color: blue;
    }
  }
\`;

      ↓ ↓ ↓ ↓ ↓ ↓

const styled = { div() {} };

const MyStyledComponent = styled.div\`
.moreSpecific & {
  background-color: yellow;
  color: {props => props.theme.color};
  
  .foo {
    border: 1px solid black;

    & .bar {
      background-color: blue;
    }
  }
}
\`;
"
`;
